### PR TITLE
Change music based on boss health

### PR DESCRIPTION
**Pull request by shanekim28**
Added a packet to go to next music phase, changed audio manager judgment system, and fixed beat clock drift on later parts of the song. Also made enemy health as a static constant for easier testing

### CODE REVIEW

File: server/prefabs/Enemy.cpp [hunk 1]
```diff
@@ -13,49 +13,84 @@
...
     this->GetComponent<NetTransform>()->SetRotation(glm::vec3(0, 0, 0));
     Collider* hitbox = new Collider(this, this->GetComponent<NetTransform>());
-    //TODO: test size values
+    // TODO: test size values
     hitbox->SetRadius(50);
     hitbox->SetHeight(20);
 
     AddComponent(hitbox);
     CollisionManager::instance().add(this);
     this->currentPhase = PHASE1;
```
Comment (Kikit24) [inline server/prefabs/Enemy.cpp]: Should there be a line to set prevPhase = PHASE1 here?

**Review by junweic01 — COMMENTED** [review]
looks good!

File: server/prefabs/Enemy.cpp [hunk 2]
```diff
@@ -64,43 +99,43 @@ void Enemy::update(float deltaTime) {
...
-    switch(this->currentPhase){
-        case PHASE1: // Swipe
-            AttackManager::instance().newSwipeAttack();
-            std::cout << "SwipeAttack!" << std::endl;
+    switch (this->currentPhase) {
+    case PHASE1: // Swipe
+        AttackManager::instance().newSwipeAttack();
+        std::cout << "SwipeAttack!" << std::endl;
+
+        this->currentPhase = PHASE2;
```
Comment (Kikit24) [inline server/prefabs/Enemy.cpp]: Delete this line since the phase logic is now implemented in update()

File: server/prefabs/Enemy.cpp [hunk 3]
```diff
@@ -64,43 +99,43 @@ void Enemy::update(float deltaTime) {
...
-        case PHASE3: // Mark / projectile
-            AttackManager::instance().newMarkedAttack();
-            std::cout << "MarkedAttack!" << std::endl;
+    case PHASE3: // Mark / projectile
+        AttackManager::instance().newMarkedAttack();
+        std::cout << "MarkedAttack!" << std::endl;
 
-            this->currentPhase = PHASE4;
-            break;
+        this->currentPhase = PHASE4;
```
Comment (Kikit24) [inline server/prefabs/Enemy.cpp]: ^

File: server/prefabs/Enemy.cpp [hunk 4]
```diff
@@ -64,43 +99,43 @@ void Enemy::update(float deltaTime) {
...
-        case PHASE4: // Laser beams
-            AttackManager::instance().newLaserAttack();
-            std::cout << "LaserAttack!" << std::endl;
+    case PHASE4: // Laser beams
+        AttackManager::instance().newLaserAttack();
+        std::cout << "LaserAttack!" << std::endl;
 
-            this->currentPhase = PHASE1;
-            break;
+        this->currentPhase = PHASE1;
```
Comment (Kikit24) [inline server/prefabs/Enemy.cpp]: ^

**Review by Kikit24 — CHANGES_REQUESTED** [review]
Looks good, nice changes. Just a few changes to clean up previous enemy phase logic to work with these improvements.

**Review by Kikit24 — APPROVED** [review]

**Pull request merged**